src/lib/security/selinux.rb

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Security
+  # Class for handling SELinux kernel params
+  class Selinux
+    Yast.import "Bootloader"
+
+    MODULE_NAME = "selinux".freeze
+    private_constant :MODULE_NAME
+
+    ENABLE_VALUE = "1".freeze
+    private_constant :ENABLE_VALUE
```

Comment:
just note: for installation we need to somehow detect that class is run first time and in such case do proposal. Probably in constructor in Mode.initial?

Reply:
If I understood it correctly yesterday, [`Yast::Bootloader.ReadOrProposeIfNeeded`](https://github.com/yast/yast-bootloader/blob/f953fef1cb876d8a7229fee0991a443fb3bd5dcf/src/modules/Bootloader.rb#L441) will do the proposal. Doesn't it?

That method is internally executed in both methods used here, [`Yast::Bootloader.kernel_param`](https://github.com/yast/yast-bootloader/blob/f953fef1cb876d8a7229fee0991a443fb3bd5dcf/src/modules/Bootloader.rb#L332) and [`Yast::Bootloader.modify_kenel_paraams`](https://github.com/yast/yast-bootloader/blob/f953fef1cb876d8a7229fee0991a443fb3bd5dcf/src/modules/Bootloader.rb#L376). So, the proposal will be done anyway. Am I wrong?

Reply:
it is proposal for boot parameters. But I am talking about selinux proposal ( similar like kdump proposal ), which according to trello card should depends on control file?

Reply:
Ah! right!